ra2ce/analysis/adaptation/adaptation.py

```diff
@@ -64,28 +68,41 @@ def execute(self) -> GeoDataFrame:
 
     def run_cost(self) -> GeoDataFrame:
         """
-        Calculate the cost for all adaptation options.
+        Calculate the unit cost for all adaptation options.
         """
-        # Open the network without hazard data
         _cost_gdf = deepcopy(self.graph_file.get_graph())
+
         for (
             _option,
             _cost,
-        ) in self.adaptation_collection.calculate_option_cost().items():
-            _cost_gdf[f"costs_{_option.id}"] = _cost
+        ) in self.adaptation_collection.calculate_options_cost().items():
+            _cost_gdf[f"{_option.id}_cost"] = _cost
+
+        # TODO: calculate link cost instead of unit cost
 
         return _cost_gdf
 
     def run_benefit(self) -> GeoDataFrame:
         """
         Calculate the benefit for all adaptation options
         """
-        return None
+        _benefit_gdf = deepcopy(self.graph_file.get_graph())
```

Comment:
Not a big fan of deepcopies, it might be wiser to just let `run_benefit` consume a graph argument `def run_benefit(self, benefit_graph: gpd.GeoDataFrame)`, then you move the deepcopy to the consumer of the method.